mysky-seed.go

```diff
@@ -0,0 +1,279 @@
+	"sneeze", "sniff", "snout", "snug", "soapy", "sober", "soccer", "soda",
+	"soggy", "soil", "solved", "sonic", "soothe", "sorry", "sowed", "soya",
+	"space", "speedy", "sphere", "spout", "sprig", "spud", "spying", "square",
+	"stick", "subtly", "suede", "sugar", "summon", "sunken", "surfer", "sushi",
+	"suture", "swept", "sword", "swung", "system", "taboo", "tacit", "tagged",
+	"tail", "taken", "talent", "tamper", "tanks", "tasked", "tattoo", "taunts",
+	"tavern", "tawny", "taxi", "tell", "tender", "tepid", "tether", "thaw",
+	"thorn", "thumbs", "thwart", "ticket", "tidy", "tiers", "tiger", "tilt",
+	"timber", "tinted", "tipsy", "tirade", "tissue", "titans", "today", "toffee",
+	"toilet", "token", "tonic", "topic", "torch", "tossed", "total", "touchy",
+	"towel", "toxic", "toyed", "trash", "trendy", "tribal", "truth", "trying",
+	"tubes", "tucks", "tudor", "tufts", "tugs", "tulips", "tunnel", "turnip",
+	"tusks", "tutor", "tuxedo", "twang", "twice", "tycoon", "typist", "tyrant",
+	"ugly", "ulcers", "umpire", "uncle", "under", "uneven", "unfit", "union",
+	"unmask", "unrest", "unsafe", "until", "unveil", "unwind", "unzip", "upbeat",
+	"update", "uphill", "upkeep", "upload", "upon", "upper", "urban", "urgent",
+	"usage", "useful", "usher", "using", "usual", "utmost", "utopia", "vague",
+	"vain", "value", "vane", "vary", "vats", "vaults", "vector", "veered",
+	"vegan", "vein", "velvet", "vessel", "vexed", "vials", "victim", "video",
+	"viking", "violin", "vipers", "vitals", "vivid", "vixen", "vocal", "vogue",
+	"voice", "vortex", "voted", "vowels", "voyage", "wade", "waffle", "waist",
+	"waking", "wanted", "warped", "water", "waxing", "wedge", "weird", "went",
+	"wept", "were", "whale", "when", "whole", "width", "wield", "wife", "wiggle",
+	"wildly", "winter", "wiring", "wise", "wives", "wizard", "wobbly", "woes",
+	"woken", "wolf", "woozy", "worry", "woven", "wrap", "wrist", "wrong",
+	"yacht", "yahoo", "yanks",
+}
+
+// PhraseToSeed will convert a seed phrase to a seed, verifying the checksum
+// along the way.
+func PhraseToSeed(phrase string) (seed Seed, err error) {
+	// Get the words from the seed.
+	words := strings.Split(phrase, " ")
+	if len(words) != 15 {
+		return seed, errors.New("seed phrase must be 15 words")
+	}
+
+	// Get the index of each word.
+	var wordInts [13]uint16
+	for i := 0; i < 13; i++ {
+		// Figure out which word this is.
+		var j uint16
+		for j = 0; j < uint16(len(myskyDictionary)); j++ {
+			if myskyDictionary[j][:3] == words[i][:3] {
+				break
+			}
+		}
+		// Check that the word exists.
+		if myskyDictionary[j][:3] != words[i][:3] {
+			return seed, errors.New("invalid seed word")
+		}
+		// Check that the word is not overlapping the version bits.
+		if i == 12 && j > 255 {
+			return seed, errors.New("13th word of seed phrase is invalid")
+		}
+		wordInts[i] = j
+	}
+
+	// Convert the words into seed. This is copied from the skynet-js
+	// implementation.
+	curByte := 0
+	curBit := 0
+	for i := 0; i < 13; i++ {
```

Comment:
we should probably pull out all the hardcoded values in this function.

It would also be nice to have some testing for all those helpers.